### PR TITLE
Add support for events on effects (added, changed, removed)

### PR DESCRIPTION
**Pull request by mzlarsson**
See other PRs as well: https://github.com/kulturkrock/screencrash-ui/pull/53 and https://github.com/kulturkrock/screencrash-core/pull/14

### CODE REVIEW

File: audio/src/audio.py [hunk 1]
```diff
@@ -1,73 +1,161 @@
...
         pg.mixer.init()
         pg.init()
         pg.mixer.set_num_channels(nof_max_sounds)
+        self._event_callback = event_callback
         self._sounds = {}
+        self._sound_infos = {}
+        self._end_events = {}
+        self._current_end_event = 1000
+
+        self._event_handle_thread = threading.Thread(target=self._event_handler)
```
Comment (AntonNeld) [inline audio/src/audio.py]: I'm always nervous when threads are involved. If I understand the flow correctly, CoreConnection._ws is accessed both from the main thread and this thread. Is this safe?

File: audio/src/audio.py [hunk 2]
```diff
@@ -1,73 +1,161 @@
...
+        self._event_handle_thread.start()
 
-    def _generate_id(self):
+    def _event_handler(self):
```
Comment (AntonNeld) [inline audio/src/audio.py]: "event" is used in multiple places, and I'm having difficulty keeping track. This one handles events from pygame, right? Maybe call it _pygame_event_handler?

File: audio/src/commandhandler.py [hunk 3]
```diff
@@ -1,25 +1,55 @@
...
     def handle_message(self, message):
         try:
             cmd = message["command"]
-            entity_id = message["entity_id"]
+            entity_id = message["entityId"]
             return self._handle_command(cmd, entity_id, message)
         except Exception as e:
             return self._create_error_msg(f"Failed to carry out command. {e}")
 
+    def _handle_event(self, event_type, entity_id):
```
Comment (AntonNeld) [inline audio/src/commandhandler.py]: _handle_mixer_event?

File: audio/src/audio.py (hunk 1 above)
Comment (mzlarsson) [reply audio/src/audio.py]: It mainly depends on whether the implementation of the websocket-client lib is thread safe. Multithread support is enabled by default and they have an info page about weird quirks found with threading (see [here](https://websocket-client.readthedocs.io/en/latest/threading.html)) and as far as I can tell it should be fine in this context. Good point though!

**Review by AntonNeld — APPROVED** [review]

**Pull request merged**